crates/interpreter/src/valid.rs

```diff
@@ -145,12 +318,19 @@ impl<'m, M: ValidMode> Context<'m, M> {
                 let t = self.functype(x as FuncIdx).unwrap();
                 let mut locals = t.params.to_vec();
                 parser.parse_locals(&mut locals)?;
-                let branch_table = Expr::check_body(self, &mut parser, &refs, locals, t.results)?;
-                side_tables.push(MetadataEntry {
-                    type_idx: self.funcs[x] as usize,
-                    parser_range: Range { start: parser_start, end: parser_start + size },
-                    branch_table,
-                });
+                let branch_table =
+                    M::next_branch_table(&mut side_table, self.funcs[x] as usize, Range {
+                        start: parser_start,
+                        end: parser_start + size,
+                    });
```

Comment:
```suggestion
                    })?;
```

In particular, we shouldn't `unwrap` below. Those are user errors, not internal invariants.